astropy/table/serialize.py

```diff
@@ -264,17 +288,19 @@ def represent_mixins_as_columns(tbl, exclude_classes=()):
 
 
 def _construct_mixin_from_obj_attrs_and_info(obj_attrs, info):
-    cls_full_name = obj_attrs.pop('__class__')
-
-    # If this is a supported class then import the class and run
-    # the _construct_from_col method.  Prevent accidentally running
-    # untrusted code by only importing known astropy classes.
-    if cls_full_name not in __construct_mixin_classes:
+    cls_full_name = obj_attrs.pop('__class__', None)
+    if cls_full_name is None:
+        cls = SerializedColumn
+    elif cls_full_name not in __construct_mixin_classes:
+        # If this is a supported class then import the class and run
+        # the _construct_from_col method.  Prevent accidentally running
+        # untrusted code by only importing known astropy classes.
         raise ValueError(f'unsupported class for construct {cls_full_name}')
+    else:
+        mod_name, cls_name = re.match(r'(.+)\.(\w+)', cls_full_name).groups()
```

Comment:
what magic is this? I see it's old code so presumably OK... but what's the regex doing?

Reply:
Looks like it just splits on the last `.`, which I think one can do with `cls_full_name.rpartition('.')` just as well...

Maybe I should add this as a separate commit, just as clean-up?

Reply:
Did add another small commit, even though it is out of scope...